arcgis-ios-sdk-samples/Utility Network/Display utility association/DisplayUtilityAssociationViewController.swift

```diff
@@ -0,0 +1,137 @@
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapType: .topographicVector, latitude: 41.8057655, longitude: -88.1489692, levelOfDetail: 23)
+            mapView.graphicsOverlays.add(associationsOverlay)
+        }
+    }
+    
+    @IBOutlet var attachmentBBI: UIBarButtonItem!
+    @IBOutlet var connectivityBBI: UIBarButtonItem!
+    
+    private let utilityNetwork = AGSUtilityNetwork(url: URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!)
+    private let maxScale = 2000.0
+    private let associationsOverlay = AGSGraphicsOverlay()
+    private let attachmentSymbol = AGSSimpleLineSymbol(style: .dot, color: .green, width: 5)
+    private let connectivitySymbol = AGSSimpleLineSymbol(style: .dot, color: .red, width: 5)
+    
+    func loadUtilityNetwork() {
+        utilityNetwork.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get all the edges and junctions in the network.
+                let sourcesByType = Dictionary(grouping: self.utilityNetwork.definition.networkSources) { $0.sourceType }
+                let operationalLayers = self.mapView.map?.operationalLayers
+                // Add all edges that are not subnet lines to the map.
+                let edgeLayers = sourcesByType[.edge]!
+                    .filter { $0.sourceUsageType != .subnetLine }
+                    .map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: edgeLayers)
+                // Add all the junctions to the map.
+                let junctionLayers = sourcesByType[.junction]!.map { AGSFeatureLayer(featureTable: $0.featureTable) }
+                operationalLayers?.addObjects(from: junctionLayers)
+                
+                // Create a renderer for the associations.
+                let attachmentValue = AGSUniqueValue(description: "Attachment", label: "", symbol: self.attachmentSymbol, values: [AGSUtilityAssociationType.attachment])
+                let connectivityValue = AGSUniqueValue(description: "Connectivity", label: "", symbol: self.connectivitySymbol, values: [AGSUtilityAssociationType.connectivity])
+                self.associationsOverlay.renderer = AGSUniqueValueRenderer(fieldNames: ["AssociationType"], uniqueValues: [attachmentValue, connectivityValue], defaultLabel: "", defaultSymbol: nil)
+                
+                // Populate the legened.
+                self.attachmentSymbol.createSwatch(withBackgroundColor: nil, screen: .main) { [weak self] (image, error) in
+                    if let error = error {
+                        print("Error creating swatch: \(error)")
+                    } else {
+                        self?.attachmentBBI.image = image?.withRenderingMode(.alwaysOriginal)
```

Comment:
I just remembered reading this in the [documentation](https://developer.apple.com/documentation/uikit/uibaritem/1616415-image):

> You should set this property before adding the item to a bar.

It would be best to follow that recommendation, if possible. That may mean waiting to create and set the toolbar items until you have both images.

Reply:
@philium I've updated the code, please let me know what you think!